scala/scala_import.bzl

```diff
@@ -1,121 +1,105 @@
-#intellij part is tested manually, tread lightly when changing there
-#if you change make sure to manually re-import an intellij project and see imports
-#are resolved (not red) and clickable
+load(":providers.bzl", "JarsToLabels")
+
+# Note to future authors:
+#
+# Tread lightly when modifying this code! IntelliJ support needs
+# to be tested manually: manually [re-]import an intellij project
+# and ensure imports are resolved (not red) and clickable
+#
+
 def _scala_import_impl(ctx):
-    target_data = _code_jars_and_intellij_metadata_from(ctx.attr.jars)
-    (current_target_compile_jars, intellij_metadata) = (target_data.code_jars, target_data.intellij_metadata)
-    current_jars = depset(current_target_compile_jars)
-    exports = _collect(ctx.attr.exports)
-    transitive_runtime_jars = _collect_runtime(ctx.attr.runtime_deps)
-    jars = _collect(ctx.attr.deps)
-    jars2labels = {}
-    _collect_labels(ctx.attr.deps, jars2labels)
-    _collect_labels(ctx.attr.exports, jars2labels) #untested
-    _add_labels_of_current_code_jars(depset(transitive=[current_jars, exports.compile_jars]), ctx.label, jars2labels) #last to override the label of the export compile jars to the current target
-    return struct(
-        scala = struct(
-          outputs = struct (
-              jars = intellij_metadata
-          ),
-        ),
-        jars_to_labels = jars2labels,
-        providers = [
-            _create_provider(current_jars, transitive_runtime_jars, jars, exports)
-        ],
+
+    direct_binary_jars = []
+    all_jar_files = []
+    for jar in ctx.attr.jars:
+        for file in jar.files:
+            all_jar_files.append(file)
+            if not file.basename.endswith("-sources.jar"):
+                direct_binary_jars += [file]
```

Comment:
Why use += here but append above? Can we use append here too?

Reply:
It's proof that I am not as consistent as I wish to be.

Reply:
I'm pretty sure that I wrote the `.append` and andy wrote the `+=`. It seems it is proof that I don't always read the surrounding code as well as I should :D

Reply:
if we're touching this, wdyt about moving it to a private method as well? same level of abstraction for the top level